src/utils/getInitialURL.js

```diff
@@ -0,0 +1,53 @@
+const DEFAULT_LAT = 40.1125785
+const DEFAULT_LNG = -88.2287926
+const DEFAULT_ZOOM = 1
+
+export const parseUrl = () => {
+  const url = window.location.href
+  const geocoordMatch = url.substring(url.indexOf('@'))
+
+  const getCoords = isValidCoord(geocoordMatch)
+
+  if (getCoords.valid) {
+    return {
+      center: { lat: getCoords.lat, lng: getCoords.lng },
+      zoom: getCoords.zoom,
+    }
+  } else {
+    return {
+      center: { lat: DEFAULT_LAT, lng: DEFAULT_LNG },
+      zoom: DEFAULT_ZOOM,
+    }
+  }
+}
+
+const isValidCoord = (geocoordMatch) => {
```

Comment:
- Try matching against one big regex
- Consider using just null as an error state
- Use if statements in sequence to catch things early
- Reduce duplicate calls (perhaps a between function?)